### DISCUSSION TITLE
Measurement Scope to specific layer.

### DISCUSSION THREAD

**Discussion by dvazquez-lightbox** [geoman-io/leaflet-geoman#1531] [Questions & Help] [answered]
Hi everyone, When using the measure tool we noticed it applies to all layers globally. Is it possible to scope it to individual layers like the other modes ex: layer.pm.enableMeasure()

**Comment by Falke-Design** [comment]
Yes you can enable the measurement with following:

```
layer.pm.getOptions().measurements.measurement = true
```

Please let me know, if it works.

**Reply by dvazquez-lightbox** [reply]
Thanks for the reply Falke, when i try the following my issue is that on the first draw the measurement is empty, for example 

```
map.on("pm:drawstart", ({ workingLayer }) => {
    console.log(workingLayer.pm.measurements) // empty object on first load, expected but now all other draws have measurement after pm:create was fired once
})

map.on("pm:create", ({ layer }) => {
    layer.pm.getOptions().measurements.measurement = true // every other subsequent draw now contains the measurement object and displays on screen tooltip 
})
```

On the first draw, the measurement is not applied, even when hovered.
Every other drawing after the first draw now has measurement enabled from drawStart. Let me know if this is the correct behavior, thank you

**Reply by Falke-Design** [reply]
Hi,
no that is not expected. Give me please some time, to figure out a workaround for you

**Reply by Falke-Design** [reply, marked as answer]
I think the best way for you is this:

```
map.on("pm:create", ({ layer }) => {
  const measurements = {
    measurement: true,
    showTooltip: true,
    showTooltipOnHover: true,
    totalLength: true,
    segmentLength: true,
    area: true,
    radius: true,
    perimeter: true,
    height: true,
    width: true,
    coordinates: true,
    displayFormat: 'metric',
  };

  layer.on('pm:enable', (e)=>{
    layer.pm.setOptions({measurements}); // enables measurement calculation
    layer.pm.addMeasurementTooltipToLayer(layer); // Adds the tooltip to the layer
  });
  layer.pm.setOptions({measurements}); // enables measurement calculation
  layer.pm.addMeasurementTooltipToLayer(layer); // Adds the tooltip to the layer
});
```

The `pm:enable` listener is need, because with enabling the global edit mode, the global options are applied to the layer (disabled measurement) and therefore you need set the new measurement options (enabled measurement).

Please let me know, if this works for you.

**Reply by dvazquez-lightbox** [reply]
Thank you for the quick response, I will get back to you and let you know if it worked.

**Reply by dvazquez-lightbox** [reply]
@Falke-Design just tried it, works great. thank you!
